articles/macos-setup-experience.md

```diff
@@ -178,6 +178,8 @@ If you configure software and/or a script for setup experience, users will see a
 
 This window shows the status of the software installations as well as the script exectution. Once all steps have completed, the window can be closed and Setup Assistant will proceed as usual.
 
+> Currently, software set to [automatic install](https://fleetdm.com/guides/automatic-software-install-in-fleet) is also installed during Setup Assistant but won't appear in the window. All software, automatic and setup experience, isn't installed in any particular order. The script always runs after setup experience software is installed.
```

Comment:
@noahtalerman this sounds right to me, but I would like to double check 

> All software, automatic and setup experience, isn't installed in any particular order

Can I get back to you tomorrow morning on this?

Reply:
@jahzielv sounds good 👍 

Heads up that I updated the language in [this commit](https://github.com/fleetdm/fleet/pull/24736#discussion_r1883981223).

Is there a good spot on the Setup experience page in the UI so IT admins are aware?

Reply:
@noahtalerman dug into the details this morning. It's a little complicated 😅 

> **TL;DR - _technically_ there is no particular order that software will be installed. In most situations though, automatic install software will install before setup experience software. I think the "no order" language is best since it's the truth/least confusing. I'll leave it up to you if we want to get more into the weeds.** 

The only order we enforce re: software installations is [here](https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software_installers.go#L28), where we list the software installations that are pending. This list, ordered by creation time, is what gets sent to Orbit so it knows what software to install.

If setup experience is supposed to be run, we currently pause _all_ software installations until after swiftDialog is installed on the host (see [here](https://github.com/fleetdm/fleet/blob/main/orbit/pkg/installer/installer.go#L92)). 

We also do not create a software installation request for setup experience software until swiftDialog is installed on the host (see [here](https://github.com/fleetdm/fleet/blob/main/orbit/pkg/setup_experience/setup_experience.go#L62)).

Downloading and installing swiftDialog can take a while, and on top of that Orbit only re-checks the config every 30s, so in practice the host's queries (including the automatic install policy queries) will have run long before swiftDialog is in place and the above checks pass.

Since pending software installations are listed in order of creation (like noted at the top), and since in practice the automatic policy installation requests will be created _before_ the setup experience installation requests (because we're waiting on swiftDialog), in practice automatic installations will run before setup experience ones.

But this order isn't enforced by code, it's just happening because typically time to download swiftDialog < time to run policies.

Reply:
> Is there a good spot on the Setup experience page in the UI so IT admins are aware?

Could we add a link to the doc updated in this PR under/next to this one? Maybe something like `Learn about how Fleet installs software on hosts`?

![Screenshot 2024-12-13 at 9 40 34 AM](https://github.com/user-attachments/assets/6343af49-f359-433d-b6ed-c8cf0d96fcee)